File: ABCActivityDemo/app/src/main/java/com/lkb/abcactivitydemo/MainActivity.java
package com.lkb.abcactivitydemo;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
Button next;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        next = findViewById(R.id.button_next_activity_a);
        final Intent i = new Intent(this, Main2Activity.class);
        //i.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_CLEAR_TOP);
        next.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
              startActivity(i);
            }
        });
    }
}

File: BaseAndroidProject/app/src/main/java/com/lkb/baseandroidproject/MainActivity.kt
package com.lkb.baseandroidproject

import android.content.Intent
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : BaseActivity(), MyDialogFragment.DialogClickListener {
    var dialogFragment: MyDialogFragment? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        toolbar.setTitle(R.string.app_name)
//        val db = Firebase.firestore
//        val city = hashMapOf(
//            "name" to "Baripada",
//            "state" to "Odisha",
//            "country" to "India",
//        )
//        val address = hashMapOf(
//            "data" to listOf(city, city)
//        )
//        db.collection("address").document("LA")
//            .set(address)
//            .addOnSuccessListener {
//                Log.d(
//                    "MainActivity",
//                    "DocumentSnapshot successfully written!"
//                )
//            }
//            .addOnFailureListener { e -> Log.w("MainActivity", "Error writing document", e) }
        logout.setOnClickListener {

            dialogFragment = MyDialogFragment()
            dialogFragment!!.setOnDialogClickListener(this)
            val fm = supportFragmentManager
            dialogFragment!!.show(fm, getString(R.string.dialog_tag))
        }
    }

    override fun onSuccess() {
        signOut()
    }

    override fun onFailure() {
        //do nothing
    }

    override fun onStop() {
        super.onStop()
        dialogFragment?.removeListener()
    }
}
